Actions/AnalyzeTests/TestResultAnalyzer.ps1

```diff
@@ -1,107 +1,324 @@
-                $appTests += [int]$suite.tests
-                $appFailed += [int]$suite.failures
-                $appSkipped += [int]$suite.skipped
-                $appTime += [decimal]::Parse($suite.time, [System.Globalization.CultureInfo]::InvariantCulture)
-            }
-            $appPassed = $appTests-$appFailed-$appSkipped
-            Write-Host "- $appName, $appTests tests, $appPassed passed, $appFailed failed, $appSkipped skipped, $appTime seconds"
-            $summarySb.Append("|$appName|$appTests|") | Out-Null
-            if ($appPassed -gt 0) {
-                $summarySb.Append("$($appPassed) :white_check_mark:") | Out-Null
-            }
-            $summarySb.Append("|") | Out-Null
-            if ($appFailed -gt 0) {
-                $summarySb.Append("$($appFailed) :x:") | Out-Null
+    if (Test-Path -Path $testResultsFile -PathType Leaf) {
+        $testResults = [xml](Get-Content -path $testResultsFile -Encoding UTF8)
+        $totalTests = 0
+        $totalTime = 0.0
+        $totalFailed = 0
+        $totalSkipped = 0
+        if ($testResults.testsuites) {
+            $appNames = @($testResults.testsuites.testsuite | ForEach-Object { $_.Properties.property | Where-Object { $_.Name -eq "appName" } | ForEach-Object { $_.Value } } | Select-Object -Unique)
+            if (-not $appNames) {
+                $appNames = @($testResults.testsuites.testsuite | ForEach-Object { $_.Properties.property | Where-Object { $_.Name -eq "extensionId" } | ForEach-Object { $_.Value } } | Select-Object -Unique)
             }
-            $summarySb.Append("|") | Out-Null
-            if ($appSkipped -gt 0) {
-                $summarySb.Append("$($appSkipped) :white_circle:") | Out-Null
+            foreach($testsuite in $testResults.testsuites.testsuite) {
+                $totalTests += $testsuite.Tests
+                $totalTime += [decimal]::Parse($testsuite.time, [System.Globalization.CultureInfo]::InvariantCulture)
+                $totalFailed += $testsuite.failures
+                $totalSkipped += $testsuite.skipped
             }
-            $summarySb.Append("|$($appTime)s|\n") | Out-Null
-            if ($appFailed -gt 0) {
-                $failuresSb.Append("<details><summary><i>$appName, $appTests tests, $appPassed passed, $appFailed failed, $appSkipped skipped, $appTime seconds</i></summary>\n") | Out-Null
+            Write-Host "$($appNames.Count) TestApps, $totalTests tests, $totalFailed failed, $totalSkipped skipped, $totalTime seconds"
+            $summarySb.Append('|Test app|Tests|Passed|Failed|Skipped|Time|\n|:---|---:|---:|---:|---:|---:|\n') | Out-Null
+            foreach($appName in $appNames) {
+                $appTests = 0
+                $appTime = 0.0
+                $appFailed = 0
+                $appSkipped = 0
+                $suites = $testResults.testsuites.testsuite | where-Object { $_.Properties.property | Where-Object { $_.Name -eq 'appName' -and $_.Value -eq $appName } }
```

Comment:
Not sure if it matters, but it looks like `extensionId` should also be supported (line 26).

Reply:
You are right